Docs/update code sec docs and workflow (#49)
uktrade/.github: SECURITY.md

`https://github.com/uktrade/REPO_NAME/settings/custom-properties`

**Mandatory**
- `reusable_workflow_opt_in` — set to `true`
- `scs_portfolio` — the portfolio associated with your Senior Civil Servant (SCS). If your portfolio is missing, this can be added by raising a ticket with the SRE team

**Optional**
- `is_docker` — for repositories that build Docker images
- `language` — select all languages used by the repository, so the organisation-level workflows run language-specific checks

#### CodeQL for Fork-Based PRs (Optional)

The DBT GitHub security configuration does not currently support scanning PRs raised from a fork of a repository. Fork PRs typically come from contributors outside the organisation, so leaving them unscanned would create a gap in coverage.

If PRs from forks must be supported, switch to [**Advanced** CodeQL](https://docs.github.com/en/code-security/how-tos/find-and-fix-code-vulnerabilities/configure-code-scanning/configuring-advanced-setup-for-code-scanning) — follow the [step-by-step instructions](https://github.com/uktrade/.github/blob/main/docs/codeql-advanced-setup.md).

---

### Organisation-Applied Controls

> An organisation administrator applies these centrally, and they cannot be weakened at repository level. The security configuration and rulesets are the mechanism; branch protection and push protection are the enforced results — verify they are active rather than configuring them yourself.

Author: ```suggestion
An organisation administrator applies these centrally, and they cannot be weakened at repository level. Verify they are active rather than configuring them yourself.
```

An organisation administrator must apply it — follow the [step-by-step instructions](https://github.com/uktrade/.github/blob/main/docs/github-security-configuration.md).

#### Branch Protection Rules

Branch protection stops unreviewed code reaching the default branch — the version of the code that gets deployed and that others build on. An organisation [ruleset](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets) has been created to apply a minimum set of branch protection rules:

- A PR is required for merges into the default branch (usually `main`)
- At least 1 approver is required before a PR can be merged
- Any conversations on the PR must be marked as resolved

> Repository administrators may add additional rules to their own repositories, but cannot weaken the organisation ruleset: where rules overlap, the most restrictive rule applies. For example, a repository ruleset that drops the required number of approvers to 0 would have no effect, while one that raises it to 3 would apply.

Author: ```suggestion
Confirm the default branch protection policy ruleset is applied under Settings → Rules → Rulesets.

Repository administrators may add additional rules to their own repositories, but cannot weaken the organisation ruleset: where rules overlap, the most restrictive rule applies. For example, a repository ruleset that drops the required number of approvers to 0 would have no effect, while one that raises it to 3 would apply.
```

#### GitHub Secret Protection

The DBT GitHub security configuration enables two complementary features. [Push protection](https://docs.github.com/en/code-security/concepts/secret-security/push-protection) blocks pushes containing high-confidence secret formats, stopping them before they reach the repository's history. [Secret scanning](https://docs.github.com/en/code-security/secret-scanning/introduction/about-secret-scanning) covers a broader range, detecting secrets already committed and raising alerts in the **Security** tab. Both can apply [custom patterns](https://docs.github.com/en/code-security/how-tos/secure-your-secrets/customize-leak-detection/define-custom-patterns) for DBT-specific secrets.

> Confirm both features are enabled on your repository. If you need additional custom patterns, raise a ticket with the SRE team.

Author: ```suggestion
Confirm both features are enabled for your repository under Settings → Advanced Security → Secret Protection. If you need additional custom patterns, raise a ticket with the SRE team.
```

#### Vulnerability Scanning

> The DBT GitHub security configuration also enables the features described under [Handling Vulnerabilities](#handling-vulnerabilities) — CodeQL, Dependabot, Dependabot security updates and dependency review. Confirm these are active on your repository.

Author: ```suggestion
The DBT GitHub security configuration and rulesets also enables the features described under [Handling Vulnerabilities](#handling-vulnerabilities) — CodeQL, Dependabot, Dependabot security updates and dependency review. Confirm these are active on your repository.
```
